File: Lv.2/12973.js
function solution(s)
{
    let stack = [s[0]]
    if (s.length % 2 != 0) return 0;
    
    for (let i=1; i < s.length; i++) {
        stack.push(s[i]);
        if (stack[stack.length - 1] == stack[stack.length - 2]) {
            stack.pop(); stack.pop();
        }
    }
    
    if (stack.length == 0) return 1

    return 0;
}

File: Lv.2/12973.py
def solution(s):
    stack = [s[0]]
    
    if (len(s) % 2 != 0): return 0

    for i in range(1, len(s)):
        stack.append(s[i])
        if (len(stack) >= 2 and stack[-1] == stack[-2]):
            stack.pop()
            stack.pop()
            
    if (len(stack) == 0): return 1
    
    return 0
